src/shared/components/pipeline-run-logs/logs/Logs.tsx

```diff
@@ -41,21 +42,37 @@ const Logs: React.FC<LogsProps> = ({
   const contentRef = React.useRef<HTMLDivElement>(null);
   const [error, setError] = React.useState<boolean>(false);
   const resourceStatusRef = React.useRef<string>(resourceStatus);
+  const blockContentRef = React.useRef<string>('');
+
   const onCompleteRef = React.useRef<(name) => void>();
   onCompleteRef.current = onComplete;
 
   const appendMessage = React.useRef<(blockContent) => void>();
 
+  const addContent = debounce(() => {
+    contentRef.current.innerText += blockContentRef.current;
+    blockContentRef.current = '';
+  }, 200);
+
+  const safeScroll = throttle(
+    () =>
+      requestAnimationFrame(() => {
+        scrollToRef.current.scrollIntoView({ behavior: 'smooth' });
+      }),
+    100,
+  );
+
   appendMessage.current = React.useCallback(
     (blockContent: string) => {
       if (contentRef.current && blockContent) {
-        contentRef.current.innerText += blockContent;
+        blockContentRef.current += blockContent;
+        addContent();
       }
       if (scrollToRef.current && blockContent && render && autoScroll) {
-        scrollToRef.current.scrollIntoView({ behavior: 'smooth' });
+        safeScroll();
```

Comment:
Why is adding content and scrolling in separate debounce / throttle ?

In ODC we do this:
```
  const addContentAndScroll = React.useCallback(
    throttle(() => {
      if (contentRef.current) {
        contentRef.current.innerText += blockContentRef.current;
      }
      if (scrollToRef.current) {
        scrollToRef.current.scrollIntoView({ behavior: 'smooth', block: 'end' });
      }
      blockContentRef.current = '';
    }, 300),
    [],
  );
```

Reply:
> Why is adding content and scrolling in separate debounce / throttle ?



This issue ([RHTAPBUGS-107](https://issues.redhat.com/browse/RHTAPBUGS-107)) exists in ODC as well, we need fix it there too.  :) 

Adding content (using `elem.innerText`) and `elem.scrollingIntoView` in same throttle call is causing too many blocking operations causing the autoscroll feature to not work and browser goes unresponsive at moments. My Idea was to limit the `scrollIntoView` call especially during the phase where the data moves rapidly, hence used debounce to append the content in batches and perform scroll operation once in 100ms using throttle.